### PR TITLE
enh(core): generate centreon gorgone configuration file

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

To introduce Gorgone daemon new module, we need to generate the configuration file.
The installation step 6 will now, generate an additional file : "gorgone.yaml"

**Fixes** # (none)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

Perform a fresh installation and check that the "gorgone.yaml" file is created correctly.

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [ ] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/install/steps/process/configFileSetup.php [hunk 1]
```diff
@@ -79,22 +80,52 @@
...
+    '/--SSLMODE--/',
+    '/--CENTREON_TRAPDIR--/',
+];
+
+$gorgoneReplacements = [
+    "var/spool/centreon",
+    "0.0.0.0",
+    8085,
+    false,
+    "etc/snmp/centreon_traps",
```
Comment (sc979) [inline www/install/steps/process/configFileSetup.php]:
```suggestion
    "/etc/snmp/centreon_traps",
```

File: www/install/steps/process/configFileSetup.php [hunk 2]
```diff
@@ -79,22 +80,52 @@
...
+$gorgonePatterns = [
+    '/--CENTREON_SPOOL--/',
+    '/--HTTPSERVERADDRESS--/',
+    '/--HTTPSERVERPORT--/',
+    '/--SSLMODE--/',
+    '/--CENTREON_TRAPDIR--/',
+];
+
+$gorgoneReplacements = [
+    "var/spool/centreon",
```
Comment (sc979) [inline www/install/steps/process/configFileSetup.php]:
```suggestion
    "/var/spool/centreon",
```

File: www/install/var/configFileGorgoneTemplate [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  db_centreon:
+    dsn: "mysql:host=--ADDRESS--:--DBPORT--;dbname=--CONFDB--"
+    username: "--DBUSER--"
+    password: "--DBPASS--"
+  db_centstorage:
+    dsn: "mysql:host=--ADDRESS--:--DBPORT--;dbname=--STORAGEDB--"
+    username: "--DBUSER--"
+    password: "--DBPASS--"
+gorgonecore:
+  privkey: "--CENTREON_SPOOL--/.gorgone/privkey.pm"
```
Comment (cgagnaire) [inline www/install/var/configFileGorgoneTemplate]: Maybe something like "rsakey.priv.pem" for the key name (like for the public one)

Comment (sc979) [reply www/install/var/configFileGorgoneTemplate]: indeed

File: www/install/var/configFileGorgoneTemplate [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    remote_dir: "--CENTREON_VARLIB--/remote-data/"
+
+  - name: engine
+    package: "gorgone::modules::centreon::engine::hooks"
+    enable: true
+    command_file: "--CENTREON_VARLIB---engine/rw/centengine.cmd"
+
+  - name: broker
+    package: "gorgone::modules::centreon::broker::hooks"
+    enable: false
```
Comment (kduret) [inline www/install/var/configFileGorgoneTemplate]: @cgagnaire should we enable it by default ?

Comment (cgagnaire) [reply www/install/var/configFileGorgoneTemplate]: For now we disable it as the "storage" of those stats files needs to change a little bit.

Comment (cgagnaire) [reply www/install/var/configFileGorgoneTemplate]: It can know.

File: www/install/var/configFileGorgoneTemplate [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    cache_dir: "--CENTREON_VARLIB--/broker-stats/"
```
Comment (cgagnaire) [inline www/install/var/configFileGorgoneTemplate]:
```suggestion
    cache_dir: "--CENTREON_CACHEDIR--/broker-stats/"
```

File: www/install/var/configFileGorgoneTemplate (hunk 4 above)
Comment (cgagnaire) [inline www/install/var/configFileGorgoneTemplate]:
```suggestion
    enable: true
```

**Comment by sc979** [comment]
thx @cgagnaire ;)

**Pull request merged**